### PR TITLE
Advanced networking settings

### PR DESCRIPTION
**Pull request by mcm001**
Removes physicalInterface from NetworkConfig, we can get it by looking it up in the list of NetworkManager interfaces instead. Exposes managed NetworkManager interface in the UI.

Add disable-networking and clear-config command line opts

Don't love the UI yet, idk.

image

### CODE REVIEW

File: photon-client/src/stores/settings/GeneralSettingsStore.ts [hunk 1]
```diff
@@ -28,10 +29,12 @@ export const useSettingsStore = defineStore("settings", {
         network: {
             ntServerAddress: "",
             shouldManage: true,
+            canManage: true,
             connectionType: NetworkConnectionType.DHCP,
             staticIp: "",
             hostname: "photonvision",
-            runNTServer: false
+            runNTServer: false,
+            networkInterfaceNames: []
```
Comment (srimanachanta) [inline photon-client/src/stores/settings/GeneralSettingsStore.ts:37]: Proper placeholder values are required, we can't just assume an empty string. (Pipeline settings are valid; other Network settings are valid)

**Review by srimanachanta — CHANGES_REQUESTED** [review]

File: photon-client/src/stores/settings/GeneralSettingsStore.ts (hunk 1 above)
Comment (mcm001) [reply photon-client/src/stores/settings/GeneralSettingsStore.ts:37]: That's an empty list of strings, no?

Comment (srimanachanta) [reply photon-client/src/stores/settings/GeneralSettingsStore.ts:37]: We should have some default values that are probable. It is better than just having an empty field which would leave people confused in the event of a no connect

File: photon-client/src/components/settings/NetworkingCard.vue [hunk 2]
```diff
@@ -14,7 +16,7 @@ const isValidNetworkTablesIP = (v: string | undefined): boolean => {
   // Check if it is a team number longer than 5 digits
   const badTeamNumberRegex = /^[0-9]{5,}$/;
 
-  if(v === undefined) return false;
+  if (v === undefined) return false;
```
[the remark this answers is not in the document]
Comment (mcm001) [reply photon-client/src/components/settings/NetworkingCard.vue:18]: Is that not something the linter does for us?

File: photon-client/src/components/settings/NetworkingCard.vue [hunk 3]
```diff
@@ -91,48 +123,75 @@ const saveGeneralSettings = () => {
...
-          :list="['DHCP','Static']"
+          :input-cols="12 - 4"
+          :list="['DHCP', 'Static']"
+          :disabled="!allowNetworkChanges()"
         />
         <cv-input
-          v-if="useSettingsStore().network.connectionType === NetworkConnectionType.Static"
           v-model="useSettingsStore().network.staticIp"
-          :input-cols="12-3"
+          :input-cols="12 - 4"
```
[the remark this answers is not in the document]
Comment (mcm001) [reply photon-client/src/components/settings/NetworkingCard.vue:131]: again, why doesn't the linter fix this for me?

File: photon-client/src/stores/settings/GeneralSettingsStore.ts (hunk 1 above)
Comment (mcm001) [reply photon-client/src/stores/settings/GeneralSettingsStore.ts:37]: Sure. Though this should probably get hidden if the list length is zero?

File: photon-client/src/stores/settings/GeneralSettingsStore.ts [hunk 4]
```diff
@@ -82,12 +85,11 @@ export const useSettingsStore = defineStore("settings", {
             this.network = data.networkSettings;
         },
         saveGeneralSettings() {
```
[the remark this answers is not in the document]
Comment (mcm001) [reply photon-client/src/stores/settings/GeneralSettingsStore.ts:90]: cursed that it's split up like that

File: photon-client/src/components/settings/NetworkingCard.vue [hunk 5]
```diff
@@ -26,60 +28,90 @@ const isValidIPv4 = (v: string | undefined) => {
...
   return hostnameRegex.test(v);
 };
 
+function allowNetworkChanges() {
+  return useSettingsStore().network.shouldManage && useSettingsStore().network.canManage;
+};
+
 const saveGeneralSettings = () => {
-  const changingStaticIp = useSettingsStore().network.connectionType === NetworkConnectionType.Static;
+  const changingStaticIp =
```
[the remark this answers is not in the document]
Comment (mcm001) [reply photon-client/src/components/settings/NetworkingCard.vue]: linter??

File: photon-client/src/stores/settings/GeneralSettingsStore.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (srimanachanta) [reply photon-client/src/stores/settings/GeneralSettingsStore.ts:93]: Probably best to handle in another  PR

File: photon-client/src/components/settings/NetworkingCard.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (srimanachanta) [reply photon-client/src/components/settings/NetworkingCard.vue:18]: The rules don't seem to be extending vue recommended for some reason, that is why I had to explicitly state a lot of them like I did. Would need to do the same for these or figure out why eslint is being difficult

**Comment by srimanachanta** [comment]
I'm confused why CI is failing?

**Comment by mcm001** [comment]
I agree it's weird. Re running

**Review by srimanachanta — APPROVED** [review]
Minus formatting fixes, everything looks fine

**Comment by mcm001** [comment]
image
thonk??

**Pull request merged**